index.src.html

```diff
@@ -4946,6 +4949,49 @@
 : <dfn event for=ImageTrack>change</dfn>
 :: Fired at the {{ImageTrack}} when the {{ImageTrack/frameCount}} is altered.
 
+Resource Reclamation{#resource-reclamation}
+==============================================
+
+When resources are constrained, a User Agent MAY proactively reclaim codecs.
+This is particularly true in the case where hardware codecs are limited, and
+shared accross all web pages.
+
+To <dfn>reclaim a codec</dfn>, a User Agent MUST run the appropriate
+close algorithm (amongst [=Close AudioDecoder=], [=Close AudioEncoder=],
+[=Close VideoDecoder=] and [=Close VideoEncoder=]) with a {{QuotaExceededError}}
+{{DOMException}}.
+
+The rules governing when a codec may be reclaimed depend on whether the codec is
+an [=active=] or [=inactive=] codec and/or a [=background=] codec.
+
+An <dfn lt="active codec|active">active codec</dfn> is a codec that has
+received a call to `encode()` or `decode()` in the past `10 seconds`, or has
+called its `output()` callback in the past `10 seconds`. An
+<dfn lt="inactive codec|inactive">inactive codec</dfn> is any codec that does
+not meet the definition of an [=active codec=].
```

Comment:
This is good, but we need to add another condition, when a frame is being encoded: it's well possible that there is 10s between calling `encode()` and the encoded frame being output. I tested encoding a 1080p file to av1 on my rather powerful Macbook with `rav1e`, a high quality encoding is > 10s per frame.

Reply:
I added a special case for VideoEncoder, and added a note.

A concern @chcunningham and I shared was "how do we reclaim stalled VideoEncoders", which is why I went with "making progress in encoding" as the definition for activity